### PR TITLE
Aarushi/onboarding

### PR DESCRIPTION
**Pull request by AarushiSingh214**
## Overview

Added the multiple onboarding screens that would allow the user to make their bio for resell. 

## Changes Made

I added two new screens, the first one allows the user to enter their name and bio and the second screen allows the user to enter their venmo if they want to. Then on the home screen a popup will appear that will allow the user to click on a get started button

## Test Coverage

Tested the feature by clicking through the various screens and making sure the code flow is right on my ios device.

## Next Steps (delete if not applicable)

Will add a button so a user can select their profile picture and will add a logout component.

## Screenshots (delete if not applicable)

<details>

  <summary>Screen Shot Name</summary>

  

https://user-images.githubusercontent.com/70385317/164817121-525bf302-db81-4583-9940-fb5748cbc716.MP4

https://user-images.githubusercontent.com/70385317/164817135-070994c5-32c8-46d3-8233-75fbed2b3f3a.MP4

</details>

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -90,7 +112,12 @@ export default function App() {
               </View>
             </View>
           )}
-          {signedIn && <Navigation colorScheme={colorScheme} />}
+
+          <Navigation
+            colorScheme={colorScheme}
+            onboard={onBoard}
+            signedIn={signedIn}
```
Comment (Haichen-Wang) [inline App.tsx:119]: this is clever!

File: components/GetStartedPullUp.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    flexDirection: 'row',
+    alignItems: "center",
+    marginBottom: 25
+  },
+  title: {
+    fontSize: 24,
+    fontWeight: "bold",
+    color: "#9E70F6",
+    marginBottom: 20
+    // alignItems: "center",
```
Comment (Haichen-Wang) [inline components/GetStartedPullUp.tsx:72]: NIT: remove this

File: components/GetStartedPullUp.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  pullUpScrollView: {
+    flex: 1,
+    width: "100%",
+    backgroundColor: 'white',
+    marginBottom: 130
+  },
+  container_header: {
+    justifyContent: "flex-start",
+    width: "100%",
+
```
Comment (Haichen-Wang) [inline components/GetStartedPullUp.tsx:45]: NIT: empty line

File: components/GetStartedPullUp.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  profileImage: {
+    width: 30,
+    height: 30,
+    borderRadius: 15
+  },
+  details: {
+    fontSize: 18,
+    paddingTop: 40,
+    paddingHorizontal: 40,
+    // paddingRight: 10,
```
Comment (Haichen-Wang) [inline components/GetStartedPullUp.tsx:91]: NIT: Remove this

File: components/PurpleButton.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
+// this is the general layout for the button that allows users to continue
+import React from "react";
+import { StyleSheet, TouchableOpacity, Text, View } from "react-native";
+
+export default function GreyButton({ text, onPress }) {
```
Comment (Haichen-Wang) [inline components/PurpleButton.tsx:5]: we need to be able to dim the button when the user has not filled out all the required info. Are you thinking about creating another component or extending the button here to accept a parameter for this?

File: components/SkipButton.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+// this is the general layout for the button that allows users to continue
```
Comment (Haichen-Wang) [inline components/SkipButton.tsx:1]: this comment is exactly the same as the one in `PurpleButton`

File: components/SkipButton.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import React from 'react';
+import { StyleSheet, TouchableOpacity, Text, View } from 'react-native';
+
+export default function SkipButton({ text, onPress }) {
+  return (
+    <TouchableOpacity onPress={onPress}>
+      <View style={styles.button}>
+        <Text style={styles.buttonText}> {"Skip"}</Text>
+
```
Comment (Haichen-Wang) [inline components/SkipButton.tsx:10]: NIT: empty line

File: navigation/LinkingConfiguration.ts [hunk 8]
```diff
@@ -34,16 +34,23 @@ const linking: LinkingOptions<RootStackParamList> = {
...
           },
         },
+
       },
 
       Modal: 'modal',
       NotFound: '*',
-      NewPost:'NewPost',
-      ProductHome: 'product/:id', // allow product details to be opened with a link specifying the product id
-      ChatWindow: 'ChatWindow'
```
Comment (Haichen-Wang) [inline navigation/LinkingConfiguration.ts:45]: why did we get rid of `ChatWindow` here

File: screens/HomeScreen.tsx [hunk 9]
```diff
@@ -16,12 +21,29 @@ import Header from "../assets/svg-components/header";
...
+  const [welcomeState, setWelcomeState] = useState(showPanel);
+
+  const intro: Intro = {
+    // images: [require("../assets/images/bluepants.png")],
+    title: "Welcome to Resell",
+    // price: 25,
+    // sellerName: "ravina patel",
+    // sellerProfile: "../assets/images/profile-pic-test.png",
+    description: "Thrifting and selling has never been this easy",
+    // similarItems: [require("../assets/images/similar-items-test.png")],
```
Comment (Haichen-Wang) [inline screens/HomeScreen.tsx:45]: remove all the comments here

File: screens/HomeScreen.tsx [hunk 10]
```diff
@@ -65,6 +87,35 @@ export default function HomeScreen({ navigation }) {
...
+                setWelcomeState(false)
+              }}
+            />
+          </View>
+
+
+          {/* <DetailPullUpHeader item={item} />
+          <DetailPullUpBody item={item} /> */}
+        </View>
+        {/* <Text style={styles.welcome}>Welcome To Resell</Text> */}
```
Comment (Haichen-Wang) [inline screens/HomeScreen.tsx:117]: remove comments here

File: screens/OnBoardScreen.tsx [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+export default function OnBoardScreen({ navigation }) {
+
+  return (
+    <View style={styles.container}>
+      <Text style={styles.titleText}>Set Up Your Profile</Text>
+      <Image
+        style={styles.profilePic}
+        source={require("../assets/images/empty_profile.png")}
+      />
+      <Text style={styles.username}>Username*</Text>
```
Comment (Haichen-Wang) [inline screens/OnBoardScreen.tsx:21]: this `*` should go away when text input is filled

File: components/GetStartedPullUp.tsx [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+  );
+}
+
+export function DetailPullUpBody({ item }: { item: Intro }) {
+  return (
+    <ScrollView style={styles.pullUpScrollView} >
+      <Text style={styles.details}>{item.description}</Text>
+    </ScrollView >
+  );
+}
```
Comment (Haichen-Wang) [inline components/GetStartedPullUp.tsx:33]: I don't think we need this right? seems like you're not using it

**Review by Haichen-Wang — CHANGES_REQUESTED** [review]
I'm very impressed by your recording! The welcome modal looks super nice and would be a huge add for the user experience! 

I am requesting changes because there're some code clean-up needed before the merge (see my comments) and also that the continue button does not appear to be dimmed when no username is entered. 

Please make the changes and re-request my review. 

Again, very great work overall!

**Review by JessieWang0706 — APPROVED** [review]

File: navigation/index.tsx [hunk 13]
```diff
@@ -103,23 +99,21 @@ const ProfileStack = createNativeStackNavigator<ProfileStackParamList>();
 
 const OnboardStack = createNativeStackNavigator<OnboardStackParamList>();
 
-function RootNavigator({ onboard, signedIn }) {
+function RootNavigator({ onboard }) {
   return (
-    <Stack.Navigator>
-      {!onboard && signedIn && (
+    <Stack.Navigator initialRouteName="ProfileOnboard">
```
Comment (Haichen-Wang) [inline navigation/index.tsx:104]: what is this used for? would this cause onboarding screen to get displayed when a user has onboarded

File: screens/LinkVenmoScreen.tsx [hunk 14]
```diff
@@ -1,16 +1,25 @@
...
+  const setOnboarded = async () => {
+    try {
+      await AsyncStorage.setItem("Onboarded", "true");
+    } catch (e) {
+      console.log(e);
+    }
+  };
+  const fromOnboard = async () => {
+    try {
+      await AsyncStorage.setItem("FromOnboard", "true");
```
Comment (Haichen-Wang) [inline screens/LinkVenmoScreen.tsx:18]: where is `FromOnboard` used?

**Review by Haichen-Wang — APPROVED** [review]
Looks great, thanks for the change! I added 2 comments but mainly for my understanding

File: navigation/index.tsx (hunk 13 above)
Comment (JessieWang0706) [reply navigation/index.tsx:104]: yes

File: screens/LinkVenmoScreen.tsx (hunk 14 above)
Comment (JessieWang0706) [reply screens/LinkVenmoScreen.tsx]: cleaned

**Pull request merged**